src/main/java/com/tagtraum/perf/gcviewer/model/GCModel.java

```diff
@@ -651,9 +651,10 @@ private void updatePromotion(GCEvent event) {
             }
 
             if (youngEvent != null) {
-                promotion.add((youngEvent.getPreUsed() - youngEvent.getPostUsed())
-                        - (event.getPreUsed() - event.getPostUsed())
-                );
+                int promo = (youngEvent.getPreUsed() - youngEvent.getPostUsed()) - (event.getPreUsed() - event.getPostUsed());
+                if (promo >= 0) {
```

Comment:
This will probably only hide the error and still not produce correct promotion numbers.